tfe/resource_tfe_workspace.go

```diff
@@ -669,3 +676,8 @@ func resourceTFEWorkspaceImporter(d *schema.ResourceData, meta interface{}) ([]*
 
 	return []*schema.ResourceData{d}, nil
 }
+
+// Warns the user that a workspace tag containing uppercase characters will be downcased.
+func warnWorkspaceTagsCasing(ctx context.Context, tag string) {
+	log.Printf("[WARN] The tag \"%s\" for contains uppercased characters that will be downcased by the API. Please update your configuration to avoid conflicts with state.", tag)
```

Comment:
```suggestion
	log.Printf("[WARN] The tag \"%s\" contains uppercase characters that will be transformed by the API. Please update your configuration to lowercase tags in order to avoid conflicts with state.", tag)
```